src/static/dataloader.py

```diff
@@ -1,56 +1,183 @@
+model_ids = {
+    "rost": {
+        "model_parameters": "1wzwVGzZPpfQbOy3fJGEHgWivfwYqBAjt",
+        "contact_matrices": "1g1CudF3RdJdEcz1T_MnSYJdRBpFIeSAz",
+        "age_distribution": "1M64F15FTN44dpjhglPY-bcNSARqpbnmT"
+    },
+    "chikina": {
+        "model_parameters": "1XDf5_Jd6Z1ao0Y6Lxeidqmha12ixCDxK",
+        "contact_matrices": "1SvSxAhbYamx_wK4ug8lK7QZsu_8TX2Dg",
+        "age_distribution": "1ULrvitaTHS73RtZQx5_I781av040yFdS"
+    },
+    "moghadas": {
+        "model_parameters": "1Ae_TLyIHsRIKEZYLuMePZshftrgYvI2T",
+        "contact_matrices": "1pLfqqUn1FGo4MBN8gCXDYrxQXHCGBfXl",
+        "age_distribution": "1bgvP9W08ZOFyjz1FN9Mj3e3G-4YzXJyZ"
+    },
+    "seir": {
+        "model_parameters": "1296mBAN2fy_R8kuXQNvvc7HEv-dSuC1S",
+        "contact_matrices": "1ryvN37HHCxNidFLbxTtW_4wFU_3owDS5",
+        "age_distribution": "1UPtjEUJPax1f6Pr1np2Y2aLz0SG2nw8n"
+    },
+    "italy": {
+        "model_parameters": "1a-kb9EpSSl0dyH6TZRXsXtomRIM1BEHH",
+        "contact_matrices": "1er8pEI1Q2gnG07PJYRxkz8PHoGC0_1_W",
+        "age_distribution": "1hvfOY5gPvvtBbfQuSmPr3FDyRnMR3LVN"
+    },
+    "kenya": {
+        "model_parameters": "1tY03EqrmcxI6iWypWZQpYyrHSxkBPJ6x",
+        "contact_matrices": "1E7ebxcj-V2aj6PPvGiEPVi_cgw1Ot2Y0",
+        "age_distribution": "11-jdcccXeHC3nfuJQ1vTL-b8fs2vKgcC"
+    },
+    "british_columbia": {
+        "model_parameters": "1NRnLBlzswN6I0xiRA97jA3yPDp04nFae",
+        "contact_matrices": "1_ju10AS0VPRDcUpmMMUYr30rkjtA7kRl",
+        "age_distribution": "11bsQqIywcEIV34zl5N1FKSRqxQ108dXC"
+    }
+}
+
+labels_file_id = "1Qe7poWcPwol0xfVIYP23Xk9SUnzHZpwI"  # Shared labels file ID
 
 PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
 
 
 class DataLoader:
-    def __init__(self):
+    def __init__(self, model: str):
+        self.model = model.lower()
+        self._validate_model()
+
+        # Set file paths and create data directory
+        self._set_file_paths()
+        self._create_data_directory()
+
+        # Download the files from Google Drive
+        self._download_data()
+
+        # Load the data
+        self.age_data = self._get_age_data()
+        self.model_parameters_data = self._get_model_parameters_data()
+        self.contact_data = self._get_contact_mtx()
+        self.labels = self._get_labels()
+
+    def _validate_model(self):
+        """Validate if the model is supported."""
+        if self.model not in model_ids:
+            raise ValueError(f"Model '{self.model}' is not supported.")
+
+    def _set_file_paths(self):
+        """Set file paths for the selected model."""
+        data_files = {
+            "model_parameters": f"{self.model}_model_parameters.json",
+            "contact_matrices": f"{self.model}_contact_matrices.xls",
+            "age_distribution": f"{self.model}_age_distribution.xls",
+            "labels": "labels.json"
+        }
+
         self._model_parameters_data_file = os.path.join(PROJECT_PATH, "../data",
```

Comment:
Break this and the following 3 definitions after `os.path.join(`. In this case, the arguments will be aligned to each other.